SimPEG/regularization/rotated.py

```diff
@@ -0,0 +1,266 @@
+                )
+            elif mesh.dim == 3:
+                anis_alpha = np.stack(
+                    (
+                        anis_alpha[..., 0, 0],
+                        anis_alpha[..., 1, 1],
+                        anis_alpha[..., 2, 2],
+                        anis_alpha[..., 0, 1],
+                        anis_alpha[..., 0, 2],
+                        anis_alpha[..., 1, 2],
+                    ),
+                    axis=-1,
+                )
+        self._anis_alpha = anis_alpha
+
+    # overwrite the call, deriv, and deriv2...
+    def __call__(self, m):
+        G = self.cell_gradient
+        M_f = self.W
+        r = G @ (self.mapping * (self._delta_m(m)))
+        return 0.5 * r @ M_f @ r
+
+    def deriv(self, m):
+        m_d = self.mapping.deriv(self._delta_m(m))
+        G = self.cell_gradient
+        M_f = self.W
+        r = G @ (self.mapping * (self._delta_m(m)))
+        return m_d.T * (G.T @ (M_f @ r))
+
+    def deriv2(self, m, v=None):
+        m_d = self.mapping.deriv(self._delta_m(m))
+        G = self.cell_gradient
+        M_f = self.W
+        if v is None:
+            return m_d.T @ (G.T @ M_f @ G) @ m_d
+
+        return m_d.T @ (G.T @ (M_f @ (G @ (m_d @ v))))
+
+    @property
+    def cell_gradient(self):
+        """The (approximate) cell gradient operator
+
+        Returns
+        -------
+        scipy.sparse.csr_matrix
+        """
+        if getattr(self, "_cell_gradient", None) is None:
+            mesh = self.regularization_mesh.mesh
+            try:
+                cell_gradient = mesh.cell_gradient
+            except AttributeError:
+                a = mesh.face_areas
+                v = mesh.average_cell_to_face @ mesh.cell_volumes
+                cell_gradient = sp.diags(a / v) @ mesh.stencil_cell_gradient
+
+            v = np.ones(mesh.n_cells)
+            # Turn off cell_gradient at boundary faces
+            if self.active_cells is not None:
+                v[~self.active_cells] = 0
+
+            dv = cell_gradient @ v
+            P = sp.diags((np.abs(dv) <= 1e-16).astype(int))
+            cell_gradient = P @ cell_gradient
+            if self.active_cells is not None:
+                cell_gradient = cell_gradient[:, self.active_cells]
+            self._cell_gradient = cell_gradient
+        return self._cell_gradient
+
+    @property
+    def _weights_shapes(self):
+        reg_mesh = self.regularization_mesh
+        mesh = reg_mesh.mesh
+        return [(mesh.n_faces,), (reg_mesh.n_cells,)]
+
+    @property
+    def W(self):
+        """The inner product operator using rotated coordinates
+
+        Returns
+        -------
+        scipy.sparse.csr_matrix
+        """
+        if getattr(self, "_W", None) is None:
+            mesh = self.regularization_mesh.mesh
+            n_faces = mesh.n_faces
+            n_cells = self.regularization_mesh.n_cells
+            cell_weights = np.ones(n_cells)
+            face_weights = np.ones(n_faces)
```

Comment:
@jcapriot : curious if those face weights will allow for sparsity? For what I gather, it won't, correct?

Reply:
They'd likely allow for it to be sparse in x-y-z orientations, but I don't think they would work properly for making it sparse in a rotated direction. It might be approximately that on minimization but I'm not sure on that front either because I haven't tried it.